File: backend/app/schemas/archive.py
from pydantic import BaseModel


class ArchiveStatusOut(BaseModel):
    state: str | None = None
    trade_date: str | None = None
    total: int = 0
    done: int = 0       # 已处理数（进度 = done/total）
    ok: int = 0         # 成功数
    failed: int = 0
    started_at: int | None = None
    finished_at: int | None = None
    error: str | None = None


class ArchiveTriggerResponse(BaseModel):
    task_id: str
    trade_date: str


class BackfillStatusOut(BaseModel):
    state: str | None = None          # running / done / error
    window: str | None = None         # "120" / "365" / "all"
    total: int = 0
    done: int = 0       # 已处理（含 failed），进度 = done/total
    ok: int = 0
    failed: int = 0
    started_at: int | None = None
    finished_at: int | None = None
    error: str | None = None


class BackfillTriggerResponse(BaseModel):
    task_id: str
    window: str

File: frontend/src/api/archive.ts
import { apiGet, apiPost } from "./client";

export interface ArchiveStatus {
  state: "running" | "done" | "error" | null;
  trade_date: string | null;
  total: number;
  done: number;
  ok: number;
  failed: number;
  started_at: number | null;
  finished_at: number | null;
  error: string | null;
}

export const triggerArchive = (date?: string) =>
  apiPost<{ task_id: string; trade_date: string }>(
    `/archive/minute${date ? `?date=${date}` : ""}`
  );

export const getArchiveStatus = () =>
  apiGet<ArchiveStatus | null>("/archive/minute/status");

export interface BackfillStatus {
  state: "running" | "done" | "error" | null;
  window: string | null;
  total: number;
  done: number;
  ok: number;
  failed: number;
  started_at: number | null;
  finished_at: number | null;
  error: string | null;
}

export const triggerChipBackfill = (days: string) =>
  apiPost<{ task_id: string; window: string }>(
    `/archive/chip-backfill?days=${days}`
  );

export const getChipBackfillStatus = () =>
  apiGet<BackfillStatus | null>("/archive/chip-backfill/status");

export const triggerDailyKlineArchive = (count = 250) =>
  apiPost<{ task_id: string; trade_date: string }>(
    `/archive/daily?count=${count}`
  );

export const getDailyKlineArchiveStatus = () =>
  apiGet<ArchiveStatus | null>("/archive/daily/status");
